FAQ and more content documentation changes (#4938)
demisto/content: docs/context_and_ouputs/README.MD

# Overview

Context is a map (dictionary) /JSON object that is created for each incident and is used to store structured results from the integration commands and automation scripts. The context keys are strings and the values can be strings, numbers, objects, and arrays.

> The main use of context is to pass data between playbook tasks, one task stores output in the context and the other task reads that output from the context and uses it.

Reviewer: maybe add a screenshot of task input/output here?

Let's look at a real world example that can show you how context is used:

For example **ThreatStream** integration has command **threatstream-analysis-report**. 
Returns the report of a file or URL that was submitted to the sandbox.

#### REST API returns
```json
{
    "Category": "File",
    "Completed": "2019-05-30 14:06:33",
    "Duration": 68,
    "Network": [
        {
            "UdpDestination": "8.8.8.8",
            "UdpPort": 53,
            "UdpSource": "192.168.2.4"
        },
        {
            "UdpDestination": "192.168.2.4",
            "UdpPort": 65324,
            "UdpSource": "8.8.8.8"
        },
        {
            "UdpDestination": "192.168.2.4",
            "UdpPort": 54896,
            "UdpSource": "8.8.8.8"
        }
    ],
    "ReportID": "413336",
    "Started": "2019-05-30 14:05:25",
    "Verdict": "Benign",
    "VmID": "",
    "VmName": ""
}
```

#### Integration YML

> In Demisto integration yml we will define outputs in the following way:

Reviewer: ```suggestion
In the integration YAML file, the command outputs are defined as such:
```

  description: The port of the UDP.
  type: String
- contextPath: ThreatStream.Analysis.Verdict
  description: The verdict of the sandbox detonation.
  type: String
```

#### In the code
```python
report_id = '413336'

response_from_api = {
    "Category": "File",
    "Completed": "2019-05-30 14:06:33",
    "Duration": 68,
    "Network": [
        {
            "UdpDestination": "8.8.8.8",
            "UdpPort": 53,
            "UdpSource": "192.168.2.4"
        },
        {
            "UdpDestination": "192.168.2.4",
            "UdpPort": 65324,
            "UdpSource": "8.8.8.8"
        },
        {
            "UdpDestination": "192.168.2.4",
            "UdpPort": 54896,
            "UdpSource": "8.8.8.8"
        }
    ],
    "ReportID": "413336",
    "Started": "2019-05-30 14:05:25",
    "Verdict": "Benign",
    "VmID": "",
    "VmName": ""
} # assume that we get this response from the service

return_outputs(
    readable_outputs=tableToMarkdown(f'Analysis Report {report_id}', [response_from_api]),
    outputs={
        'ThreatStream.Analysis(val.ReportID == obj.ReportID)': response_from_api
    },
    raw_response=response_from_api
)
```

> **Note**: The code **must** match the yml! The brand name (e.g. `ThreatStream`), the object name (e.g. `Analysis`)

Reviewer: not sure its clear whats youre trying to say here

### DT (Demisto Transform Language)

> In the above example, we observe the entry context using ```(val.Address==obj.Address)```. This works to *tie together* related entry context objects. In this instance, we are using the value of the Address key as the unique identifier to search through the existing context and link related objects. This prevents data from being overwritten as well as further enriches an existing entry with more information. Learn more about linking context [here](https://github.com/demisto/content/blob/master/docs/DT).

Reviewer 2: ```suggestion
In the above example, we observe the entry context using ```(val.ReportID == obj.ReportID)```. This works to *tie together* related entry context objects. In this instance, we are using the value of the ReportID key as the unique identifier to search through the existing context and link related objects. This prevents data from being overwritten as well as further enriches an existing entry with more information. Learn more about linking context [here](https://github.com/demisto/content/blob/master/docs/DT).
```
